app/MMB/src/scenes/tickets/boardingPasses/FlightFromTo.js

```diff
@@ -10,57 +10,98 @@ import idx from 'idx';
 import AppleWallet from './appleWallet/AppleWallet';
 import DownloadButton from './DownloadButton';
 import type { FlightFromTo as RouteStopType } from './__generated__/FlightFromTo.graphql';
+import WalletContext, { type Segment } from '../../../context/WalletContext';
 
-type Props = {|
-  +data: RouteStopType,
+type PropsWithContext = {|
+  ...Props,
+  +addSegment: (segment: Segment) => void,
 |};
 
-export const FlightFromTo = (props: Props) => {
-  const date = idx(props.data, _ => _.departure.localTime);
-  const shortDate = date
-    ? DateFormatter(new Date(date)).formatToShortDate()
-    : '';
-  const time = date ? DateFormatter(new Date(date)).formatToTime() : '';
-  return (
-    <View style={styles.row}>
-      <View style={styles.dateContainer}>
-        <Text style={styles.dateText}>
-          <Translation passThrough={shortDate} />
-        </Text>
-        <Text style={styles.dateText}>
-          <Translation passThrough={time} />
-        </Text>
-      </View>
-      <View style={styles.rightColumn}>
-        <View style={[styles.row, styles.cityContainer]}>
-          <Text style={styles.cityText}>
-            <Translation
-              passThrough={idx(props.data, _ => _.departure.airport.city.name)}
-            />
+export class FlightFromTo extends React.Component<PropsWithContext> {
+  componentDidMount = () => {
+    const id = idx(this.props.data, _ => _.id) || '';
+    const airlineLogoUrl = idx(this.props.data, _ => _.airline.logoUrl) || '';
+    const flightDate = idx(this.props.data, _ => _.departure.localTime) || null;
+    this.props.addSegment({
```

Comment:
Maybe `addSegmentToContext` to make it clear that we are doing something outside of this component?

Reply:
I think since it comes from props, it should be clear that it is something passed to the component, also since it is passed from the context, ain't it clear enough?

If not, I could change it.

Reply:
I guess you're right, no need to add that.